### DISCUSSION TITLE
OAauth authorization using loopback redirect_uri

### DISCUSSION THREAD

**Discussion by brunokim** [community/community#48383] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

I'm developing a Github OAuth app and have registered the callback URL `https://init-science.uc.r.appspot.com/authorize` in the app's settings.

The documentation is explicit in that I should be allowed to use a [loopback redirect](https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/authorizing-oauth-apps#loopback-redirect-urls) instead of a subdirectory of the registered callback URL. I attempted to use this for my local development server, but it fails with a `redirect_url_mismatch` error. I've encoded the redirect URI `http://127.0.0.1/authorize` as below:

```none
https://github.com/login/oauth/authorize?client_id=284365b88beb9cafacbc&scope=user%2Crepo&redirect_uri=http%3A%2F%2F127.0.0.1%2Fauthorize&state=VGB8ZqAfmCLjBkXYdo5Bja9YI1TtgtIiMAPCQ8im6hQ

 -- redirects to -->

https://init-science.uc.r.appspot.com/authorize?error=redirect_uri_mismatch&error_description=The+redirect_uri+MUST+match+the+registered+callback+URL+for+this+application.&error_uri=https%3A%2F%2Fdocs.github.com%2Fapps%2Fmanaging-oauth-apps%2Ftroubleshooting-authorization-request-errors%2F%23redirect-uri-mismatch&state=VGB8ZqAfmCLjBkXYdo5Bja9YI1TtgtIiMAPCQ8im6hQ
```

Am I doing something wrong here, or have I found a bug in Github's OAuth implementation/documentation?

**Comment by ammmir** [comment]
I'm seeing the same behavior here with the loopback redirect. As a temporary workaround, I've created an additional OAuth app for local development.

**Comment by gigamonkey** [comment]
Is this still a thing? I seem to be also seeing it.

**Comment by brunokim** [comment]
Hey, folks, sorry I didn't update this sooner.

I've contacted Github support in March 2023 and they said that this is indeed the expected flow for OAuth apps. However, they _are_ working to support multiple callback URLs, which even got the support agent confused because they already had access to it and saw a different UI than the one available for us.

Hopefully they launch it sooner rather than later :crossed_fingers: In the meantime, I've setup a different app for dev testing that redirects to localhost.

**Comment by mysandesh** [comment]
Useful tip above - create an additional OAuth app for local development. Thanks.

**Comment by hpsin** [comment]
Hey folks, I'm sorry I didn't see this til I searched for it. 

Loopback isn't enabled by default, you need to register it as a redirect URI before we'll redirect to it. As of this week we now support multiple redirect URIs for OAuth apps: https://github.blog/changelog/2026-08-14-multiple-redirect-uris-and-token-refresh-for-oauth-apps/

Hopefully this can help the next person to find this thread.
